### DISCUSSION TITLE
[Question] How do I setup a HTTPS server?

### DISCUSSION THREAD

**Discussion by Namelessys** [daurnimator/lua-http#200] [General]
Hey there, 

I'm not into HTTPS/TLS or encryption at all, so my question might be a bit stupid.
But how do I setup a HTTPS server using lua-http?

Thx in advance.

**Comment by daurnimator** [comment]
There are examples in the repository: https://github.com/daurnimator/lua-http/blob/master/examples/server_hello.lua

**Reply by Namelessys** [reply]
I see, thx for the answer!

But once more question, how do i use a certificate?
Like, where to put it in and such?

**Reply by daurnimator** [reply]
See the docs at https://daurnimator.github.io/lua-http/0.4/#http.server.listen which will point you to https://daurnimator.github.io/lua-http/0.4/#http.server.new

In general, you will need to pass in an openssl context:
> .ctx (context object): An openssl.ssl.context object to use for tls connections. If nil is passed, a self-signed context will be generated.

Use https://daurnimator.github.io/lua-http/0.4/#http.tls.new_server_context as a base, then you'll need to [consult the luaossl documentation](https://github.com/wahern/luaossl/releases/download/rel-20190731/luaossl.pdf).

**Reply by Namelessys** [reply]
Alright. It works now.
Thx for answering my dumb questions. Very appreciate it!
